x-pack/plugins/dashboard_enhanced/public/services/drilldowns/dashboard_to_dashboard_drilldown/collect_config.tsx

```diff
@@ -16,28 +17,48 @@ export interface CollectConfigContainerProps extends CollectConfigProps {
 export const CollectConfigContainer: React.FC<CollectConfigContainerProps> = ({
   config,
   onConfig,
-  params: { savedObjects },
+  params: { getSavedObjectsClient },
 }) => {
-  const [dashboards] = useState([
-    { id: 'dashboard1', title: 'Dashboard 1' },
-    { id: 'dashboard2', title: 'Dashboard 2' },
-    { id: 'dashboard3', title: 'Dashboard 3' },
-    { id: 'dashboard4', title: 'Dashboard 4' },
-  ]);
+  const [dashboards, setDashboards] = useState([]);
+  const [searchString, setSearchString] = useState();
+  const [isLoading, setIsLoading] = useState(false);
 
   useEffect(() => {
-    // TODO: Load dashboards...
-  }, [savedObjects]);
+    setIsLoading(true);
+    getSavedObjectsClient().then(savedObjectsClient => {
+      savedObjectsClient
+        .find({
+          type: 'dashboard',
+          search: searchString ? `${searchString}*` : undefined,
+          // todo search by id
+          searchFields: ['title^3', 'description'],
+          defaultSearchOperator: 'AND',
+          perPage: 100,
+        })
+        .then(({ savedObjects }) => {
+          const dashboardList = savedObjects.map(
+            (savedObject: SimpleSavedObject<{ id: string; title: string }>) => ({
+              id: savedObject.id,
+              title: savedObject.attributes.title,
+            })
+          );
+          setDashboards(dashboardList);
```

Comment:
Let's not forget to handle race condition here. It could be that this results are no longer relevant because new search already completed